src/helper/helper.js

```diff
@@ -150,6 +161,59 @@ class DataLayerHelper {
     merge_(this.model_, /** @type {!Object<*>} */ (this.dataLayer_[0]));
   }
 
+
+  /**
+   * Register a function to respond to events with a certain name called by
+   * the command API by storing it in a map. The function will be called an
+   * time the commandAPI is called with first parameter the string name.
+   *
+   * @param {string} name The string which should be passed into the command API
+   *     to call the processor.
+   * @param {function(*):!Object} processor The function to register to be
+   *     called later.
+   *     So long as this function is not an arrow function, it can access the
+   *     abstract model interface by using the this keyword. It is recommended
+   *     not to modify the state within the function using this.set.
+   *     Changes to the model should only be achieved by the return value, a
+   *     dict whose values will automatically be merged into the model.
+   * @this {DataLayerHelper}
+   */
+  registerProcessor(name, processor) {
+    if (!(name in this.commandProcessors_)) {
+      this.commandProcessors_[name] = [];
+    }
+    this.commandProcessors_[name].push(processor);
+  }
+
+  /**
+   * Applies the given command to the value in the dataLayer with the given key.
+   * If a processor for the command has been registered, the processor function
+   * will be invoked with any arguments passed in.
+   *
+   * @param {!Array<*>} args The arguments object containing the command
+   *     to execute and optional arguments for the processor.
+   * @return {!Array<!Object<*>>} states The updates requested to
+   *     the model state, in the order they should be processed.
+   * @private
+   */
+  processArguments_(args) {
+    // Run all registered processors associated with this command.
+    const states = [];
+    const name = args[0];
+    if (this.commandProcessors_[name]) {
+      // Cache length so as not to run processors registered
+      // by other processors after the call.
+      const length = this.commandProcessors_[name].length;
+      for (let i = 0; i < length; i++) {
+        const method = this.commandProcessors_[name][i];
+        states.push(method.apply(this.abstractModelInterface_,
+            [].slice.call(args, 1)));
```

Comment:
Can the spread operator be used here?

Reply:
It adds a 1k polyfill to the output. https://docs.google.com/document/d/1XIhihp4elYAa9cFqJL9vg3f7u4kbsoZqIQa5ytBqYg0/edit#heading=h.o8b60qe0380k

Reply:
I switched this method to use spread operators (https://pastebin.com/8vDbJ1VG) and went to 2655 bytes vs 2243. There is a polyfill but does not seem to be as big as the file claimed. Nonetheless, I will leave the code as is to be safe.